src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -1342,6 +1345,15 @@ - (nullable NSNumber *)neededReadPrivilegeForClusterID:(NSNumber *)clusterID att
     return nil;
 }
 
+#ifdef DEBUG
++ (void)forceLocalhostAdvertisingOnly
+{
+    auto interfaceIndex = chip::Inet::InterfaceId::PlatformType(kDNSServiceInterfaceIndexLocalOnly);
+    auto interfaceId = chip::Inet::InterfaceId(interfaceIndex);
+    chip::app::DnssdServer::Instance().SetInterfaceId(interfaceId);
+}
+#endif // DEBUG
+
```

Comment:
forceLocalHostAdvertisingOnly` is not controller specific right ? If so, can we just make an helper class somewhere that will only be called/used for testing ? We won't need the `#ifdef DEBUG` then ?

I don't want to block this PR on that comment but it is unclear to me why it needs to be added to `MTRDeviceController`.

Reply:
You're right, that's not controller-specific.  I just wanted to hang it off of somewhere.  Note that it's not an instance method; it's a class method.

If we had a class, it would need to be there in the API #ifdef DEBUG, right?  Using a selector for this lets us not mention this anywhere in the Matter headers.

Though we could have a project header that this code includes, I guess...

Reply:
I was either thinking about:
 - directly dump the code in the test file ?
 - or add a category/extension such as

```
// MTRDeviceController+TestExtensions.h
#import "MTRDeviceController.h"

@interface MTRDeviceController (TestExtensions)
- (void) forceLocalHostAdvertisingOnly;
@end
```
and
```
//MTRDeviceController+TestExtensions.mm
#import "MTRDeviceController+TestExtensions.h"

@implementation MTRDeviceController (TestExtensions)
- (void) forceLocalHostAdvertisingOnly {
  ...
}
@end
```

and then `#import MTRDeviceController+TextExtensions.h` in the test file ?

Reply:
The goal was to not add anything to framework headers for this, even `#ifdef DEBUG`, since framework clients can well define that themselves as part of their normal build.

That means it more or less needs to be a hidden (i.e. not declared in a header) selector on something.  I chose a class selector on MTRDeviceController to emphasize that (1) this affects the behavior of controllers and (2) it's not specific to any particular controller.

We probably _could_ use a project header with some free function for setting this (can't be a class, because it needs to work in Objective-C for tests).  But all the other bits like this that we have seem to use hidden selectors.